### DISCUSSION TITLE
Granularité des statistiques régionales niveau PAT

### DISCUSSION THREAD

**Discussion by alemangui** [betagouv/ma-cantine#986] [Idées]
Les chargés de mission veulent avoir l'info à ce niveau là, mais plusieurs soucis là-dessus : 
- Le PAT n'apporte pas de garantie de longévité ni de taille
- Plus de 33000 communes en France
- Inégalité importante de taille de PATs

**Comment by hfroot** [comment]
[django-francedata](https://github.com/entrepreneur-interet-general/django-francedata) pourrait rendre l'import de communes dans notre site rapide, mais:
- ça va ouvrir la discussion de si on écrit un migration pour changer notre modèle maintenant avec les régions et départements comme un choix du texte vers une relation foreign key
- il y a toujours la question UX de comment choisir tous les communes intéressant
- on devra investiguer les cantines qui n'ont pas une code INSEE valide pour améliorer la validité de nos données (mais p-e on doit faire ça quand même)

Combien de communes en moyen a-t-il dans un PAT ?

**Reply by alemangui** [reply]
Ça change énormément c'est le souci aussi :( Il y a de PATs avec très peu de communes, et d'autres avec plus de 500. Du coup en partant d'une sélection par commune ça peut être assez embêtant pour l'utilisateur

**Comment by ghost** [comment]
Besoin d'un utilisateur (B. L. C., le 11/02) : avoir comme granularité, non le PAT, mais l'agglomération.

"Comme les agglos sont celles aujourd'hui qui donnent les grandes orientations de l'alimentation, si elles peuvent avoir des chiffres à l'échelle de leur territoire, je pense qu'elles vont se jeter sur Ma Cantine. À mon avis, c'est l'élément qui va faire qu'elles vont se mettre à utiliser massivement ce site."

**Reply by hfroot** [reply]
Les agglos de communes ou qqch d'autre ?

**Reply by ghost** [reply]
Les agglos au sens les villes (d'après Wikipedia, il y a 4 types d'agglomérations, selon leur taille : les métropoles, les communautés urbaines, les communautés d'agglomération et les communautés de communes).

**Reply by ghost** [reply]
Besoin d'un autre utilisateur (G. F., le 17/03) : avoir l'échelle "Agglo" dans l'outil Stats

**Reply by Greta712** [reply]
Besoin également exprimé par des utilisateurs. Avoir la possibilité de faire des stats au niveau "commune"

**Comment by ghost** [comment]
Besoin exprimé par un utilisateur lors d'un webinaire (29/06) : avoir des stats précises à l'EPCI près. L'EPCI (établissement public de coopération intercommunale) est une structure regroupant plusieurs communes. C'est donc un niveau de découpage géographique qui se trouve en-dessous du département et au-dessus de la commune.

**Comment by ghost** [comment]
Besoin exprimée par une utilisatrice en PAT lors d'un atelier de présentation de la plateforme (26/07) : avoir des stats précises à la commune, l'EPCI ou le PAT près.

**Reply by valeriemerle** [reply]
dans le module stat peut on avoir un champ de recherche par EPCI apres selection du département par ex? Il existe 1254 EPCI en France soit 13 en moyenne par département. La liste des EPCI est grée par INSEE

**Reply by Greta712** [reply]
je viens d'avoir une nouvelle demande dans ce sens par un chargé de projet PAT ce matin même

**Comment by ghost** [comment]
@valeriemerle, voici la discussion.
